### PR TITLE
refactored authentication into a restcontroller

### PR DESCRIPTION
**Pull request by rixlabs**
Let's open the pull request. :)

As you can see I created a controller that manage the authentication part.

The functionality is the same but I find it more readable.

In the SecurityConfig the creation of the bean is also for readability.

Let me know what do you think.

### CODE REVIEW

**Comment by akirasosa** [comment]
@rixlabs 
It looks really nice refactoring. 👍  Please let me write a few comments for more improvements.

**Comment by rixlabs** [comment]
Need access to my repo or you merge first?

File: src/main/java/com/myapp/auth/SecurityConfig.java [hunk 1]
```diff
@@ -32,6 +32,12 @@ public SecurityConfig(UserService userService,
         this.tokenAuthenticationService = tokenAuthenticationService;
     }
 
+    @Bean
+    public StatelessAuthenticationFilter authenticationTokenFilterBean() throws Exception {
+        StatelessAuthenticationFilter authenticationTokenFilter = new StatelessAuthenticationFilter(tokenAuthenticationService);
+        return authenticationTokenFilter;
+    }
```
Comment (akirasosa) [inline src/main/java/com/myapp/auth/SecurityConfig.java:39]: @rixlabs 
How about extract this bean configuration into another configuration class? For example..

``` java
@Configuration
public class StatelessAuthenticationFilterConfig {

    @Bean
    public StatelessAuthenticationFilter authenticationTokenFilterBean(TokenAuthenticationService tokenAuthenticationService) throws Exception {
        return new StatelessAuthenticationFilter(tokenAuthenticationService);
    }

}
```

Then, SecurityConfig does not need TokenAuthenticationService anymore. Instead, directly inject StatelessAuthenticationFilter.
https://github.com/springboot-angular2-tutorial/boot-app/pull/5/files#diff-678eff096639de26bd3e2aa487ffcc80L25

File: src/main/java/com/myapp/controller/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        Authentication authentication =  authenticationManager.authenticate(loginToken);
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        UserDetails userDetails = userDetailsService.loadUserByUsername(params.getEmail().get());
+
+        String token = tokenHandler.createTokenForUser(userDetails);
+
+        // Return the token
+        return ResponseEntity.ok().header("x-auth-token",token).body("");
```
Comment (akirasosa) [inline src/main/java/com/myapp/controller/AuthController.java:56]: @rixlabs 
Now, old method can be deleted.
https://github.com/springboot-angular2-tutorial/boot-app/blob/c7a594bdaa940c8994264b80839656a7370004fc/src/main/java/com/myapp/auth/TokenAuthenticationService.java#L23-L23

**Review by akirasosa — COMMENTED** [review]
@rixlabs 
Please take a look at my two review comments.

**Comment by rixlabs** [comment]
About defining the bean in another class it could be good looking from encapsulation side but it would create another file for the security configuration that is used only in SecurityConfig class. So it's good for separating concerns but add another file to the config. For me it can stay inside SecurityConfig but is up to you.

For the method addAuthentication of TokenAuthenticationService I will delete it

**Comment by rixlabs** [comment]
I commited the change but I forgotted to clean the imports.

I'm doing also that part now

**Comment by akirasosa** [comment]
@rixlabs 
Great. Merged. Thanks!

**Comment by rixlabs** [comment]
I found a bad BUG about the authenticationManager!!!!

I'll fixit and send another commit. 

I'm sorry forgot to test with anonymous

**Pull request merged**